src/kmer_unordered_set.cpp

```diff
@@ -68,50 +96,75 @@ void KmerSets::get_parent_kmer_sets(){
     path relative_hap2_kmer_list_path = "data/hg04.all.homo.unique.kmer.1000.fa";
     path absolute_hap2_kmer_list_path = project_directory / relative_hap2_kmer_list_path;
 
-    load_file_into_unordered_set(absolute_hap1_kmer_list_path,hap1_kmer_set);
-    load_file_into_unordered_set(absolute_hap2_kmer_list_path,hap2_kmer_set);
+    load_file_into_unordered_set(absolute_hap1_kmer_list_path,hap1_kmer_set,"hap1");
+    load_file_into_unordered_set(absolute_hap2_kmer_list_path,hap2_kmer_set,"hap2");
+
+}
+
+
+void KmerSets::print_component_parent_conf_matrix() {
+	cout << "Number of graph components: " << component_map.size() << endl;
+	
+	for (int i = 0; i < component_map.size(); i++) {
+
+		cout << "component: " << i << "\n   pat\tmat" << endl;
+		
+		for (int j = 0; j < 2; j++) {
+			cout << j << " |" << component_map[i][j][parent_hap1_int] << "\t|" << component_map[i][j][parent_hap2_int] << "\t|" <<endl;
+		}
+		cout << endl;
+	}
+}
 
+void KmerSets::parse_path_string(string path_string){
+	int string_delim = path_string.find('-');
+	this->graph_component = stoi(path_string.substr(0,string_delim));
+	this->component_haplotype = stoi(path_string.substr(string_delim+1,path_string.length()));
 }
 
 // single kmer find
-bool KmerSets::find_haplotype_single_kmer_count(string child_kmer) {
-	hap1_kmer_count = 0;
-	hap2_kmer_count = 0;
+bool KmerSets::increment_parental_kmer_count( string path_hap_string, string child_kmer) {
+
+	// this is done for each kmer - consider moving it out of this function
+	parse_path_string(path_hap_string);
 
 	if (hap1_kmer_set.find(child_kmer) != hap1_kmer_set.end()){
-		hap1_kmer_count++;
+		component_map[graph_component][component_haplotype][parent_hap1_int]++;
 		}
 
 	if (hap2_kmer_set.find(child_kmer) != hap2_kmer_set.end()){
-		hap2_kmer_count++;		
+		component_map[graph_component][component_haplotype][parent_hap2_int]++;	
 		}
 
-	cout << " Checking single kmer: " << child_kmer << endl;
-	cout << " found " << hap1_kmer_count << " hap1 (hg03) kmers." << endl ;
-	cout << " found " << hap2_kmer_count << " hap2 (hg04) kmers." << endl << endl ;
 	return 0;
 }
 
-// kmer set find
-bool KmerSets::find_haplotype_kmer_set_count(unordered_set <string> child_kmers) {
-	unordered_set <string> :: iterator itr;
-	hap1_kmer_count = 0;
-	hap2_kmer_count = 0;
+void KmerSets::normalize_kmer_counts(){
+	// divide by # of kmers for each parent
 
+	cerr << "Normalize Component Map..\n";
+	for (int i = 0; i < component_map.size(); i++) {		
+		for (int j = 0; j < 2; j++) {
+			float raw_count = component_map[i][j][parent_hap1_int];
+			component_map[i][j][parent_hap1_int] = (raw_count/num_hap1_kmers);
+
+			raw_count = component_map[i][j][parent_hap2_int];
+			component_map[i][j][parent_hap2_int] = (raw_count/num_hap2_kmers);
+
+		}
+	}
+}
+
+bool KmerSets::find_haplotype_kmer_set_count(string path_hap_string, unordered_set <string> child_kmers) {
```

Comment:
for "path_hap_string", i would use the same name that libbdsg uses, which is just "path_name"